snippets/core/system-text-json/csharp/PersonConverterWithTypeDiscriminator.cs

```diff
@@ -40,16 +40,16 @@ public override Person Read(ref Utf8JsonReader reader, Type typeToConvert, JsonS
                 throw new JsonException();
             }
 
-            Person value;
+            Person person;
             TypeDiscriminator typeDiscriminator = (TypeDiscriminator)reader.GetInt32();
             switch (typeDiscriminator)
```

Comment:
This would be a great use for the pattern matching switch expression:

```csharp
Person person = typeDiscriminator switch
{
    TypeDiscriminator.Customer => new Customer();
    TypeDiscriminator.Employee => new Employee();
    _ => throw new JsonException();
};
```

Reply:
Done, except I had to comma-separate the lines instead of terminating them with semicolon. 😄 

BTW, this pattern doesn't seem to be in the [C# guide switch article](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/switch) , but it is in the [C# 8 what's new](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8).